farm/modeling/language_model.py

```diff
@@ -162,30 +162,68 @@ def load(cls, pretrained_model_name_or_path, n_added_tokens=0, language_model_cl
 
         return language_model
 
-    @classmethod
-    def get_language_model_class(cls, model_name_or_path):
+    @staticmethod
+    def get_language_model_class(model_name_or_path):
         # it's transformers format (either from model hub or local)
         model_name_or_path = str(model_name_or_path)
-        if "xlm" in model_name_or_path and "roberta" in model_name_or_path:
-            language_model_class = 'XLMRoberta'
-        elif 'roberta' in model_name_or_path:
-            language_model_class = 'Roberta'
-        elif 'codebert' in model_name_or_path.lower():
+
+        config = AutoConfig.from_pretrained(model_name_or_path)
+        model_type = config.model_type
+        if model_type == "xlm-roberta":
+            language_model_class = "XLMRoberta"
+        elif model_type == "roberta":
+            if "mlm" in model_name_or_path.lower():
+                raise NotImplementedError("MLM part of codebert is currently not supported in FARM")
+            language_model_class = "Roberta"
+        elif model_type == "camembert":
+            language_model_class = "Camembert"
+        elif model_type == "albert":
+            language_model_class = "Albert"
+        elif model_type == "distilbert":
+            language_model_class = "DistilBert"
+        elif model_type == "bert":
+            language_model_class = "Bert"
+        elif model_type == "xlnet":
+            language_model_class = "XLNet"
+        elif model_type == "electra":
+            language_model_class = "Electra"
+        elif model_type == "dpr":
+            if config.architectures[0] == "DPRQuestionEncoder":
+                language_model_class = "DPRQuestionEncoder"
+            elif config.architectures[0] == "DPRContextEncoder":
+                language_model_class = "DPRContextEncoder"
```

Comment:
What abou dpr reader? 
Should we also fall back to string matching if the architecture doesnt match?

Reply:
DPR Reader is not supported at the moment (see https://github.com/deepset-ai/haystack/issues/440). Should we add a NotImplementedError?